pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -90,6 +99,28 @@ func createInferenceGraphPodSpec(graph *v1alpha1api.InferenceGraph, config *Rout
 		}
 	}
 
+	// If auth is enabled for the InferenceGraph:
+	// * Add --enable-auth argument, to properly secure kserve-router
+	// * Add the --inferencegraph-name argument, so that the router is aware of its name
+	// * Enable auto-mount of the ServiceAccount, because it is required for validating tokens
+	// * Set a non-default ServiceAccount with enough privileges to verify auth
+	if graph.GetAnnotations()[constants.ODHKserveRawAuth] == "true" {
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, "--enable-auth")
+
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, "--inferencegraph-name")
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, graph.GetName())
+
+		podSpec.AutomountServiceAccountToken = proto.Bool(true)
```

Comment:
I do remember some security discussion around auto-mounting the service account token, IIRC was about this conf added as default on KServe. 
Does it need to be enabled here?

Reply:
The automount to be configurable? Well... I don't think auth can work without the SA token mounted.

Reply:
yes, it my be required in the future by prodSec, as they already mentioned it for KServe.

Reply:
maybe, open a follow-up to see if this can be achieved without it?

Reply:
May be this may already qualify as what is requested in the other ticket?

The Jira for InferenceServices is this one: https://issues.redhat.com/browse/RHOAIENG-19904. It looks like the concern is that the token is **always mounted** even if unused. In contrast, in this PR the token is **mounted only when auth is enabled**. I know we are using the "automount" feature, but it is not "always on" as in the ISVC case.

If you think we must fully stop using the automount, I can create a Jira.

Reply:
yes, ok.